tests/unhappypath/unhappypath_test.go

```diff
@@ -0,0 +1,281 @@
+	var testrailID = 3526435
+	injectionType := "drop"
+	//TODO need to fix this issue later.
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/35264
+	var runID int
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+	It("Inject network error while applications are running", func() {
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			logrus.Infof("Iteration number %d", i)
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("applicationscaleup-%d", i))...)
+		}
+		currentTime := time.Now()
+		timeToExecuteTest := time.Now().Local().Add(time.Hour*time.Duration(totalTimeInHours) +
+			time.Minute*time.Duration(0) +
+			time.Second*time.Duration(0))
+
+		// Set Autofs trim
+		currNode := node.GetWorkerNodes()[0]
+		err := Inst().V.SetClusterOpts(currNode, map[string]string{
+			"--auto-fstrim": "on",
+		})
+		if err != nil {
+			err = fmt.Errorf("error while enabling auto fstrim, Error:%v", err)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		Step("Verify applications after deployment", func() {
+			for _, ctx := range contexts {
+				ValidateContext(ctx)
+			}
+		})
+		Step("Create snapshot schedule policy", func() {
+			createSnapshotSchedule(contexts)
+		})
+
+		for int64(timeToExecuteTest.Sub(currentTime).Seconds()) > 0 {
+			// TODO core check
+			logrus.Infof("Remaining time to test in minutes : %d ", int64(timeToExecuteTest.Sub(currentTime).Seconds()/60))
+			Step("Set packet loss on random nodes ", func() {
+				//Get all nodes and set eth0
+				nodes := node.GetWorkerNodes()
+				numberOfNodes := int(math.Ceil(float64(0.40) * float64(len(nodes))))
+				selectedNodes := nodes[:numberOfNodes]
+				//nodes []Node, errorInjectionType string, operationType string,
+				//dropPercentage int, delayInMilliseconds int
+				logrus.Infof("Set network error injection")
+				Inst().N.InjectNetworkError(selectedNodes, injectionType, "add", dropPercentage, delayInMilliseconds)
+				logrus.Infof("Wait %d minutes before checking px status ", errorPersistTimeInMinutes/(time.Minute))
+				time.Sleep(errorPersistTimeInMinutes)
+				hasPXUp := true
+				for _, n := range nodes {
+					logrus.Infof("Check PX status on %v", n.Name)
+					err := Inst().V.WaitForPxPodsToBeUp(n)
+					if err != nil {
+						hasPXUp = false
+						logrus.Errorf("PX failed to be in ready state  %v %s ", n.Name, err)
+					}
+				}
+				if !hasPXUp {
+					Expect(fmt.Errorf("PX is not ready on on or more nodes ")).NotTo(HaveOccurred())
+				}
+				logrus.Infof("Clear network error injection ")
+				Inst().N.InjectNetworkError(selectedNodes, injectionType, "delete", 0, 0)
+				//Get kvdb members and
+				if injectionType == "drop" {
+					injectionType = "delay"
+				} else {
+					injectionType = "drop"
+				}
+			})
+			logrus.Infof("Wait 20 minutes before checking px status ")
```

Comment:
Is this value based on some logic, if not, then please define it as a constant at the top of the file. That would make it easier to change in the future, w/o looking at the code in detail.